### ISSUE TITLE
`<xf:trigger appearance="minimal">` is rendered with `role="button"`, but can't be activated by pressing space

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4583]
TL;DR:

Legend: 2 issues marked with ❌

| | Key to activate | Reachable with *tab* on Safari
| - | - | -
| **Links** | 👍 Enter | 👍 No
| **Buttons** | 👍 Enter + ❌ Space | ❌ Yes

Proposed solution: for `<xf:trigger appearance="minimal">`, generate `<label role="button" tabindex="0">` instead of `<a role="button" tabindex="0" href="#">`

---

According to [MDN on `role="button"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Accessibility_concerns), users must be activate buttons with *space* (in addition to *enter*), and we don't support that with the `<xf:trigger appearance="minimal">`, which is rendered with `role="button"`. We can solve this in at least a couple of ways:

1. Adding Scala.js code that runs when *space* is pressed while a `<xf:trigger appearance="minimal">` has the focus, that prevents the default behavior, and does what would have happened if *enter* had been pressed.
2. Don't produce an anchor, and instance produce a `span`. (And don't produce the `href="#"`.)

I like this second option because it also solves navigation with tab on Safari: on Safari, by default, tab goes through form fields and buttons but not links (see notes 1 and 2). The problem is that right now, tabbing will skip buttons if they are rendered with `<xf:trigger appearance="minimal">`, but it does not skip buttons that are rendered with, say, `<label role="button" tabindex="0">`.

This is inconsistent, and, I think, not what users expect. Generating a `<label role="button" tabindex="0">` instead of a `<a role="button" tabindex="0" href="#">` would also solve this problem (see note 3).

Notes:

1. You can press option-tab if you also want to go through links.
2. This especially makes sense with a form: you might have a number of links for navigation "around" a form, but when interacting with the form, you are more likely to want to go through the form fields, rather than through the links "around" the form.
3. If we decide to fix this issue with an option other than 2, we could solve the tabbing issue by adding support for `xxf:element` on `xf:trigger`, and if the value of `xxf:element` is different than `a`, not producing the `href` attribute.
